reana_client/validation/compute_backends.py

```diff
@@ -0,0 +1,172 @@
+            workflow_steps = workflow["specification"]["steps"]
+            return SerialComputeBackendValidator(
+                workflow_steps=workflow_steps, supported_backends=supported_backends
+            )
+        if workflow_type == "yadage":
+            workflow_steps = workflow["specification"]["stages"]
+            return YadageComputeBackendValidator(
+                workflow_steps=workflow_steps, supported_backends=supported_backends
+            )
+        if workflow_type == "cwl":
+            workflow_steps = workflow.get("specification", {}).get("$graph", workflow)
+            return CWLComputeBackendValidator(
+                workflow_steps=workflow_steps, supported_backends=supported_backends
+            )
+        if workflow_type == "snakemake":
+            workflow_steps = workflow["specification"]["steps"]
+            return SnakemakeComputeBackendValidator(
+                workflow_steps=workflow_steps, supported_backends=supported_backends
+            )
+
+    workflow = reana_yaml["workflow"]
+    validator = build_validator(workflow)
+    validator.validate()
+    display_message(
+        "Workflow compute backends appears to be valid.",
+        msg_type="success",
+        indented=True,
+    )
+
+
+class ComputeBackendValidatorBase:
+    """REANA workflow compute backend validation base class."""
+
+    def __init__(
+        self,
+        workflow_steps: Union[List[Dict], None] = None,
+        supported_backends: Union[List[str], None] = [],
+    ):
+        """Validate compute backends in REANA workflow steps.
+
+        :param workflow_steps: list of dictionaries which represents different steps involved in workflow.
+        """
+        self.workflow_steps = workflow_steps
+        self.supported_backends = supported_backends
+        self.messages = []
+
+    def validate(self) -> None:
+        """Validate REANA workflow compute backend."""
+        try:
+            self.validate_compute_backends()
+        except ComputeBackendValidationError as e:
+            display_message(str(e), msg_type="error", indented=True)
+            sys.exit(1)
+
+    def validate_compute_backends(self) -> None:
+        """Validate compute backends in REANA workflow."""
+        raise NotImplementedError
+
+
+class SerialComputeBackendValidator(ComputeBackendValidatorBase):
+    """REANA serial workflow compute backend validation."""
+
+    def validate_compute_backends(self) -> None:
+        """Validate compute backends in REANA serial workflow."""
+        for step in self.workflow_steps:
+            backend = step.get("compute_backend")
+            if backend and backend not in self.supported_backends:
+                raise ComputeBackendValidationError(
+                    f'Compute backend "{backend}" found in step "{step.get("name")}" is not supported. '
+                    f'List of supported compute backends: "{", ".join(self.supported_backends)}"'
+                )
```

Comment:
Minor: I observe that this message is practically repeated in all the validators, maybe we can extract it into `ComputeBackendValidatorBase`?